src/discussions/post-comments/PostCommentsView.test.jsx

```diff
@@ -687,13 +693,131 @@ describe('ThreadView', () => {
+
+      expect(await waitFor(() => screen.findByText('No responses', { exact: true })))
+        .toBeInTheDocument();
+      expect(container.querySelector('.comments-sort'))
+        .not
+        .toBeInTheDocument();
+    });
+
+    it('should have only two options', async () => {
+      renderComponent(discussionPostId);
+
+      await waitFor(() => screen.findByTestId('comment-comment-1'));
+      await act(async () => {
+        fireEvent.click(screen.getByRole('button', { name: /Oldest first/i }));
+      });
+      const dropdown = await waitFor(() => screen.findByTestId('comment-sort-dropdown-modal-popup'));
+
+      expect(dropdown)
+        .toBeInTheDocument();
+      expect(await within(dropdown)
+        .getAllByRole('button'))
+        .toHaveLength(2);
+    });
+
+    it('should be selected Oldest first and auto focus', async () => {
+      renderComponent(discussionPostId);
+
+      await waitFor(() => screen.findByTestId('comment-comment-1'));
+      await act(async () => {
+        fireEvent.click(screen.getByRole('button', { name: /Oldest first/i }));
+      });
+      const dropdown = await waitFor(() => screen.findByTestId('comment-sort-dropdown-modal-popup'));
+
+      expect(dropdown)
+        .toBeInTheDocument();
+      expect(within(dropdown)
+        .getByRole('button', { name: /Oldest first/i }))
+        .toBeInTheDocument();
+      expect(within(dropdown)
+        .getByRole('button', { name: /Oldest first/i }))
+        .toHaveFocus();
+      expect(within(dropdown)
+        .getByRole('button', { name: /Newest first/i }))
+        .not
+        .toHaveFocus();
+    });
+
+    test('successfully handles sort state update', async () => {
+      renderComponent(discussionPostId);
+
+      expect(store.getState().comments.sortOrder)
+        .toBeFalsy();
+
+      await waitFor(() => screen.findByTestId('comment-comment-1'));
+      await act(async () => {
+        fireEvent.click(screen.getByRole('button', { name: /Oldest first/i }));
+      });
+      const dropdown = await waitFor(() => screen.findByTestId('comment-sort-dropdown-modal-popup'));
+      await act(async () => {
+        fireEvent.click(within(dropdown)
+          .getByRole('button', { name: /Newest first/i }));
+      });
+
+      expect(store.getState().comments.sortOrder)
+        .toBeTruthy();
+    });
+    test('successfully handles tour state update', async () => {
+      const tourData = [{
+        id: 15,
+        tourName: 'not_responded_filter',
+        showTour: false,
+        enabled: true,
+        user: 8,
+      }, {
+        id: 16,
+        tourName: 'response_sort',
+        showTour: false,
+        enabled: true,
+        user: 8,
+      }];
+      await store.dispatch(fetchUserDiscussionsToursSuccess(tourData));
+      renderComponent(discussionPostId);
+      await waitFor(() => screen.findByTestId('comment-comment-1'));
+      expect(selectTours(store.getState()).find(item => item.tourName === 'response_sort').enabled)
```

Comment:
find tour object and store it in a constant. That is how you do not need to find the same object multiple times.